### ISSUE TITLE
[v2] translate page not showing 

### ISSUE THREAD

**Issue by almogdesign** [gadicc/meteor-messageformat#124]
Hi, 
When I go to translate I'm getting the following 

![screen shot 2015-05-12 at 11 48 00 am](https://cloud.githubusercontent.com/assets/169507/7583330/04f572f8-f89d-11e4-9c0b-e977c76deba0.png)

**Comment by gadicc** [comment]
I think I see what's going on here.  This should fix:

```bash
$ meteor update
$ meteor
$ meteor shell # in another window
msgfmt.forceExtract(); // or just put this code in a file to run just once and then remove
$ meteor deploy ...
```

Can you let me know how that goes?

**Comment by gadicc** [comment]
no wait, sorry, redeploy won't fix this yet.  hold on :)  (but you can do all the other steps in the meantime)

**Comment by almogdesign** [comment]
@gadicc when I run msgfmt.forceExtract();

I get 

> msgfmt.forceExtract();
TypeError: Object #<Object> has no method 'forceExtract'
    at repl:1:8
    at /Users/macbookpro/GoPlatfarm/RnD/App/platfarm-app/.meteor/local/build/programs/server/shell-server.js:243:23
>

**Comment by gadicc** [comment]
can you double check that the meteor update worked and you're on core preview.8 and extract preview.6?  but actually i suggest you wait (despite what I said earlier), i'm doing a better mechanism to get the native strings from dev to production on deploy. i'm hoping to have it done today but have a few other prior commitments; if not today then tomorrow for sure.

**Comment by almogdesign** [comment]
No meteor update did not work, I think it does not update specific versions (@=)
Only thing that was updated 

harrison:papa-parse  upgraded from 1.0.10 to 1.1.0
meteorhacks:kadira   upgraded from 2.20.2 to 2.20.4

Sounds good will wait, please update me when it's ready

**Comment by gadicc** [comment]
Yeah as per last time you can remove the `@=` for any msgfmt:* stuff and just use `@`.

Ok, it's still the versions per above.  I was hoping to do an easier/automated way to get the extracted/native strings to production, but for now you just have to save `server/mfAll.js` on devel and then redeploy each time.  I've amended the instructions, try it out and see how you go.  I'll figure out something better for the future.

**Comment by gadicc** [comment]
ok, i found a creative solution, that works great, but i won't have a chance to finish it up tonight.  if you haven't touched this yet, wait till tomorrow and it will be very easy... no need to worry about mfAll.js.

**Comment by almogdesign** [comment]
Great haven't updated yet

**Comment by gadicc** [comment]
Great, I think we're all ready to go!

```bash
$ meteor update
$ meteor
$ meteor shell # in another window
msgfmt.forceExtract(); // or just put this code in a file to run just once and then remove
$ meteor deploy ...
```

For the meteor update, you want:

```
msgfmt:core                  2.0.0-preview.10
msgfmt:extract               2.0.0-preview.7
msgfmt:ui                    2.0.0-preview.2
```

If you still have an `@=` for any of these in your `.meteor/packages`, you should remove it before trying to update.

Note, the instructions above are just once-off to fix the problem from the old versions.  You won't have to worry about this again.

**Comment by gadicc** [comment]
P.S.  Not sure where you're hosting but this does rely on `NODE_ENV=="production"`.  If you look on the server logs (on production), you should see some lines like:

```
debug:  [msgfmt]  [Q] addNative() called before init(), queueing...
[Wed May 13 2015 16:10:29 GMT+0000 (UTC)] INFO 2015-05-13 16:10:29.602
debug:  [msgfmt]  addNative() called
```

Of course, it's easier to just confirm when all your strings pop up in `/translate` :)

**Comment by almogdesign** [comment]
@gadicc meteor update is not updating any of the packages?

**Comment by almogdesign** [comment]
I updated to the latest package versions and started the meteor server but I'm getting the following error and app i snot running  

 (STDERR) 
W20150514-10:19:21.552(3)? (STDERR) /Users/macbookpro/GoPlatfarm/RnD/App/platfarm-app/.meteor/local/build/programs/server/packages/msgfmt_extract.js:252
W20150514-10:19:21.552(3)? (STDERR)       fs.mkdir(dir, saveExtracts);                                            
W20150514-10:19:21.552(3)? (STDERR)                     ^
W20150514-10:19:21.562(3)? (STDERR) ReferenceError: saveExtracts is not defined
W20150514-10:19:21.562(3)? (STDERR)     at packages/msgfmt:extract/extract.js:233:1
W20150514-10:19:21.563(3)? (STDERR)     at Object.cb [as oncomplete] (fs.js:169:19)

**Comment by gadicc** [comment]
Damn, these last minute code cleanups always get me.  You should only get that once?  Otherwise try creating a `private` folder in your app root, and this should go away.

**Comment by gadicc** [comment]
Oops, forget the "only once" comment, but creating `private` will fix.

**Comment by almogdesign** [comment]
I tried re running the server and it still happened, let me try creating a private folder

**Comment by almogdesign** [comment]
So I did everything I ran msgfmt.forceExtract(); I got the following message 

'Forcing (re)-extract of all files and keys.  See results in main console log, according to your default debug level'

Deployed and still get the blank page / message

**Comment by almogdesign** [comment]
But it's only on deployment when I run on local host everything is fine. Let me try to deploy one more time.

**Comment by gadicc** [comment]
does your deployment environment set NODE_ENV to "production", and did you see anything about `addNative()` in your production server logs?  like in https://github.com/gadicc/meteor-messageformat/issues/124#issuecomment-101937226?

**Comment by almogdesign** [comment]
Yes I have the following set 
process.env.NODE_ENV // 'production'
NODE_ENV production

Not seeing addNative()

**Comment by almogdesign** [comment]
I have mfAll.js in my server folder could that be causing the issue

**Comment by gadicc** [comment]
wait where/how are you setting that?
in your app code is too late... it needs to be in your environment?
no, mfAll.js is fine... it's not exporting the latest extracted strings.  i had to get a bit tricky to make it both easy *and* not slow down the dev environment.

**Comment by gadicc** [comment]
just going to double check on my side that nothing broke, brb.

**Comment by almogdesign** [comment]
Yes environment

**Comment by almogdesign** [comment]
Just tried another deploy still noting

**Comment by gadicc** [comment]
Yeah working on something a little more robust, hang on :)

**Comment by almogdesign** [comment]
no problem

**Comment by gadicc** [comment]
I'm pretty sure we're good, but meteor deploys are down, so I can't test 100%.  But everything gets packed correctly with "meteor bundle".  Not sure what deployment strategy you're using.

You can delete `private/extract*` (or I guess in your case, the whole `private` directory), and then I hope for the very last time, use the same set of instructions:

```bash
$ meteor update
$ meteor
$ meteor shell # in another window
msgfmt.forceExtract(); # or just put this code in a file to run just once and then remove
$ meteor deploy ...
```

For the meteor update, you want:

```
msgfmt:core                  2.0.0-preview.11
msgfmt:extract               2.0.0-preview.8
msgfmt:ui                    2.0.0-preview.2
```

If you still have an `@=` for any of these in your `.meteor/packages`, you should remove it before trying to update.

Note, the instructions above are just once-off to fix the problem from the old versions.  You won't have to worry about this again.

**Comment by almogdesign** [comment]
"If you still have an @= for any of these in your .meteor/packages, you should remove it before trying to update." 

You mean just do meteor add msgfmt:core ?

**Comment by gadicc** [comment]
No, I meant in `.meteor/packages`.  But if you want to use command line, I think `meteor add msgfmt:core@2.0.0-preview.11` will also remove the `@=` version lock, otherwise you might have to `meteor remove` it first.

**Comment by almogdesign** [comment]
Everything is working thanks

**Comment by gadicc** [comment]
Amazing!  Thanks for your patience on this one.  I'm glad we found a good way to do this without slowing down the dev environment *and* without requiring extra effort by the dev.
